lib/foreman_azure_rm/engine.rb

```diff
@@ -34,11 +34,18 @@ class Engine < ::Rails::Engine
     end
 
     config.to_prepare do
+
+      # pry here and implement .class_eval here
       require 'azure_mgmt_resources'
       require 'azure_mgmt_network'
       require 'azure_mgmt_storage'
       require 'azure_mgmt_compute'
 
+      ::Image.class_eval do
+        validates :uuid, format: { with: /\A((marketplace|custom|gallery):\/\/)([A-Za-z_\-]+)[\^:A-Za-z_\-]+[\^:A-Za-z_\-]+[\^:A-Za-z_\-]+/,
+          message: "Incorrect UUID format" }
+      end
```

Comment:
added validation for prefix here. although, I was wondering if the pattern could be used even in a better way.

Reply:
This looks OK. You can avoid the `class_eval`, but that's not critical:
``` ruby
::Image.validates(:uuid, format: { with: /\A((marketplace|custom|gallery):\/\/)[^:]+(:[^:]+:[^:]+:[^:]+)?\z/,
          message: "Incorrect UUID format" })
```